webapp/components/survey/NodeDefDetails/TaxonProps.js

```diff
@@ -89,25 +80,31 @@ const TaxonProps = (props) => {
       </FormItem>
 
       <div className="taxon-props__panel-right">
-        {(showTaxonomyPanel || editedTaxonomy) && (
+        {showTaxonomyPanel && (
           <PanelRight
             width="100vw"
             onClose={() => {
-              dispatch(TaxonomyActions.setTaxonomyForEdit(null))
               setShowTaxonomyPanel(false)
             }}
             header={i18n.t('taxonomy.header')}
           >
-            <TaxonomyView showClose={false} />
+            <TaxonomyDetails showClose={false} onTaxonomyCreated={onTaxonomySelect} />
           </PanelRight>
         )}
-        {!editedTaxonomy && showTaxonomiesPanel && (
+        {!showTaxonomyPanel && showTaxonomiesPanel && (
```

Comment:
why not to have just one variable panel which can be null || taxonomy || taxonomies ?

Reply:
i will think tomorrow about that